Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM python:3.7
+
+RUN apt-get update
+RUN apt-get install --yes --no-install-recommends wget git maven less
+
+ENV JAVA_HOME /usr/lib/jvm/default-java/
+RUN mkdir /root/.gnupg \
+ && chmod 600 /root/.gnupg
+
+ADD src/requirements.txt /code/
+RUN pip3 install -r /code/requirements.txt
+
+ADD src/*.py /code/
```

Comment:
`requirements.txt` is usually in the project root. Same for the `.gitignore`. If you do that you can just add the entire `src` folder without having to filter for python files

Reply:
> `requirements.txt` is usually in the project root. Same for the `.gitignore`.

True! I'd argue here the root of the Python project is the `src` folder though.

> If you do that you can just add the entire `src` folder without having to filter for python files

That means that the root of the Python project is now the root of the repo. In that setup, `src` is a Python module, which is... just wrong :D I'm inclined to either move everything out of `src` (might make the most sense), or leave this as is.

Thoughts?

Reply:
> I'd argue here the root of the Python project is the `src` folder though.
Unless you're planning to add java or other code in the future, the root is just `apache-release-verification/`.

> That means that the root of the Python project is now the root of the repo. In that setup, `src` is a Python module, which is... just wrong
hmm, technically it's not unless you add a `__init__.py` file. And if you add it it'll be even now I think since sys/path will point to `apache-release-verification/`

Reply:
Oh I see what you mean. Yeah I didn't mean keeping the `src/` folder necessarily, mostly just being able to do `src/*` since that'll also copy over possible `.sh` files we might add in the future

Reply:
> Unless you're planning to add java or other code in the future, the root is just `apache-release-verification/`.

Exactly. I originally put all the Python stuff into `src` thinking there might be other stuff in this repo at some point. It's turning out to be only a Python script though at this point plus packaging, so I'll go ahead and get rid of `src`. Can always restructure if the need arises later.

Reply:
Hmm. On second thought.

I didn't want to add `src/*` because in the current setup that'll also add `src/venv` and `src/__pycache__` and stuff like that. Which, as you point out, shouldn't be there anyway, but in the root of the project.

Currently my favorite idea is moving everything but `.py` files out of `src`, changing instructions to build the `venv` in the project root, and add `src/*` in the `Dockerfile`.

Reply:
> Currently my favorite idea is moving everything but `.py` files out of `src`, changing instructions to build the `venv` in the project root, and add `src/*` in the `Dockerfile`.

++

Reply:
Meh, `__pycache__` gets created in `src` by Python. It contains `.pyc` files, which might be problematic when running on a macOS host. Guess this means we keep the `src/*.py`, but I'll still move the rest of the stuff out. It feels cleaner now that I thought about it.

Reply:
The usual way to deal with that is to add a .dockerignore file and blacklist __pycache__ there.

i.e. https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/container_engine/django_tutorial/.dockerignore

Reply:
Oh wait, that's what `.dockerignore` is for. Never mind.